Tara i2

tarads443 opened the pull request.

SwatiPareek commented on functions/common.py:
```diff
@@ -24,3 +25,40 @@ def mod(dividend: float, divisor: float) -> float:
         dividend += divisor
     return dividend
+
+
```
I created a file called display_output in the PR for our rad->deg conv function. 
I was thinking maybe all of these types of functions should be in that file. 
What do you think?

tarads443 replied:
Are you referring to the factorial() function or the decimal_to_binary_integer() and decimal_to_binary_decimal() functions?

EmanuelUnity3D commented:
I guess the built-in conversions weren't allowed for this project? Everything looks good to me. It just needs to be refactored to meet PEP-8 standards. Also, do we include the tests now or later?

tarads443 commented:
Hey so I changed the code to conform to PEP-8 and I added unit tests. 

Two things I'll add: 
- I'll move the functions over to output_display()
- Need to figure out how to add unit tests for negative numbers. 

I'll do this in my second push because I want to push this PR as it's a bit old and I'll create a new one for the two items above. 

Do you mind seeing if you are okay to approve?

Emancus09 commented on functions/common.py:
```diff
@@ -28,6 +29,60 @@ def factorial(argument: int) -> int:
     return result
 
 
+def mod(dividend: float, divisor: float) -> float:
+    # TODO not sure on the typing for this,
+    # and should check if can be optimized
+    while dividend > divisor:
+        dividend -= divisor
+    while dividend < divisor:
+        dividend += divisor
+    return dividend
```
I would just use the built-in mod because it's a built-in operator. Unless Python's mod works differently from this one?

Emancus09 commented on functions/common.py:
```diff
@@ -27,8 +27,8 @@ def mod(dividend: float, divisor: float) -> float:
 
-# Function to convert a whole number to binary
 def decimal_to_binary_integer(value):
+    """Returns a string that contains the binary result of an integer"""
```
The docstring convention that we adopted documents the parameters, return values, and possible exceptions thrown in addition to the description of the function itself. I'm not really partial to one convention over another but I would add those things just for the sake of consistency.

tarads443 replied:
Thanks I'll change this. Can I change this in the next PR as it's a minor change?

EmanuelUnity3D replied:
Sure as long as we remember it.

Emancus09 commented on tests/test_common.py:
```diff
@@ -20,6 +20,9 @@ def test_factorial_float_raise_error(self):
         with self.assertRaises(the_exception.InputError):
             common.factorial(56.3)
 
+    def test_factorial_float_raise_error(self):  # 0b is removed from bin result for all unit tests
+        self.assertEqual((common.convert_to_binary(6)), bin(6).replace("0b", ""))  # Test for positive integer
+        self.assertEqual((common.convert_to_binary(-6)), bin(-6).replace("0b", ""))  # Test for negative integer
```
How come we aren't testing floating point values?

tarads443 replied:
This is something that I need to add but was going to do it in the next PR as I have to figure out how because the bin() function doesn't accept floating point values.

EmanuelUnity3D replied:
Ah ok
